yunwoo/programmers-0/옹알이-(1).js

```diff
@@ -0,0 +1,26 @@
+function solution(babbling) {
+  let result = 0;
+
+  babbling.forEach((el) => {
+    const reg = /aya|ye|woo|ma/g;
+    const babblingReplace = el.replace(reg, "1");
+
+    if (
+      babblingReplace === "1" ||
+      babblingReplace === "11" ||
+      babblingReplace === "111" ||
+      babblingReplace === "1111"
+    ) {
+      result++;
+    }
+  });
+
+  return result;
+}
```

Comment:
```suggestion
function solution(babbling) {
  let result = 0;

  babbling.forEach((el) => {
    const reg = /aya|ye|woo|ma/g;
    const babblingReplace = el.replace(reg, "");

    if (babblingReplace === "") result++;
  });

  return result;
}
```
- 풀이잘봤습니다🙇‍♂️  replace 할 때  `1`로 안바꾸고 그냥 `""`로 변경해서 조건문을 단순화해서 풀이하실 수 있습니다!!!
- 아래는 로직은 같지만 reduce를 이용한 풀이입니다 :) 
```js
function solution(babbling) {
  return babbling.reduce((acc, cur) => {
    const regex = /aya|ye|woo|ma/g;
    const replacedBabbling = cur.replace(regex, "");

    return replacedBabbling === "" ? acc + 1 : acc;
  }, 0);
}
```

Reply:
- 와 그렇군요..! 지저분한 조건식이 필요하지않았네요!!!
- 참고풀이도 살펴보겠습니다 감사합니다🙇‍♂️👍